index.bs

```diff
@@ -218,7 +218,7 @@ following steps:
  <li>If |request|'s [=request/client=] is not null, then for each [=client hints token=] |requestHint| in
 |request|'s [=environment settings object/client hints set=], [=set/append=] |requestHint| to
 |hintSet|.
- <li>For each |possibleHint| in |hintSet|, if |request| is a [=subresource request=] and the result of
+ <li>For each |possibleHint| in |hintSet|, if |request| is a [=subresource request=] or a [=non-subresource request=] (except "sharedworker" and "worker") and the result of
```

Comment:
OK, so we're trying to capture a non-navigation request (captured here as subresource or non-subresource request) with any valid [`[=destination=]`](https://fetch.spec.whatwg.org/#concept-request-destination) except for "<code>sharedworker</code>" or "<code>worker</code>", right?

Reply:
Exactly

Reply:
This is still confusing to me because a request whose destination is "iframe" will match both the non-subresource [definition](https://fetch.spec.whatwg.org/#non-subresource-request) and the navigation [definition](https://fetch.spec.whatwg.org/#navigation-request), and there's no other qualifier to use to distinguish between the two. Maybe that's something to fix up in the Fetch spec?

It seems like we might want to tie this in to when a request's client is null. For example, I'd still expect permission policy to not be applied when an iframe navigates from a.com --> b.com when embedded in c.com where c.com has set a permission policy like `permission-policy: ch-ua=()`.

Reply:
Let's move this up to an early exit condition at the top

Reply:
I also clarified the worker section to only apply to service workers.